spirv_cross.cpp

```diff
@@ -3008,3 +3041,43 @@ void Compiler::analyze_variable_scope(SPIRFunction &entry)
 		this->get<SPIRVariable>(loop_variable.first).loop_variable = true;
 	}
 }
+
+uint64_t Compiler::get_buffer_block_flags(const SPIRVariable &var)
+{
+	auto &type = get<SPIRType>(var.basetype);
+
+	// Some flags like non-writable, non-readable are actually found
+	// as member decorations. If all members have a decoration set, propagate
+	// the decoration up as a regular variable decoration.
+	uint64_t base_flags = meta[var.self].decoration.decoration_flags;
+	uint64_t all_members_flag_mask = 0;
+	for (uint32_t i = 0; i < uint32_t(type.member_types.size()); i++)
+		all_members_flag_mask |= ~get_member_decoration_mask(type.self, i);
```

Comment:
I feel like this would be slightly easier to understand if mask started as ~0ull and this used &= - that way the comment above explicitly reflects what the code is doing